src/backend/utils/adt/agtype.c

```diff
@@ -7817,6 +7817,95 @@ Datum age_sqrt(PG_FUNCTION_ARGS)
     PG_RETURN_POINTER(agtype_value_to_agtype(&agtv_result));
 }
 
+PG_FUNCTION_INFO_V1(age_cbrt);
+
+Datum age_cbrt(PG_FUNCTION_ARGS)
+{
+
+    // check if input is agtype
+    if (PG_ARGISNULL(0))
+        PG_RETURN_NULL();
+
+    /*  
+        *  cbrt() only supports agtype float and 
+        *  agtype numeric and agtype integer for the input expression.
+    */
+    agtype *agtype_in = AG_GET_ARG_AGTYPE_P(0);
+    agtype_value agtv;
+    agtype_value agtv_result;
+
+    /* Variable to store result */
+    float8 result;
+
+    /* Check If input is NULL */
+    if (agtype_in == NULL)
+        PG_RETURN_NULL();
+
+
+    /* Check if input is scalar type or not */
+    if (!agtype_extract_scalar(&agtype_in->root, &agtv) ||
```

Comment:
Checkout get_float_compatible_arg, it will simplify all this, generally speaking, you should not do any casting in the function itself. Especially here, because Postgres changed float8in_internal_null in pg15 and this will add extra steps to PG15 support

Reply:
For, get_float_compatible_arg I would need to change the argument from agtype to 'variadic"any" ', to cater the paremeters for get_float_compatible_arg .

Reply:
Well I can use get_float_compatible_arg, and when I get the `type`, I can make a check there if the input is agtype or not.